addon/components/sl-menu.js

```diff
@@ -453,194 +239,386 @@ export default Ember.Component.extend({
      *
      * @function
-     * @returns {Boolean}
+     * @throws {ember/Error}
+     * @returns {undefined}
      */
-    displayShowAll: Ember.computed(
-        'isRoot',
-        'showAllBoolean',
-        function() {
-            return this.get( 'isRoot' ) && this.get( 'showAllBoolean' );
+    selectParent() {
+        const selections = this.get( 'selections' );
+
+        if ( selections.length <= 1 ) {
+            warn( '`selectParent` triggered with no parent context' );
+        }
+
+        const currentItem = Ember.get( selections.popObject(), 'item' );
+
+        if ( !currentItem ) {
+            throw new Ember.Error( 'Invalid last menu item' );
         }
-    ),
+
+        Ember.set( currentItem, 'selected', false );
+    },
 
     /**
-     * Send the primary action
+     * Select the previous sibling in the current context
      *
      * @function
+     * @throws {ember/Error}
      * @returns {undefined}
      */
-    performAction() {
-        this.$().addClass( 'active' );
+    selectPrevious() {
+        const selections = this.get( 'selections' );
 
-        const rootNode = this.get( 'rootNode' );
-        rootNode.sendAction( 'selectionMade', this.get( 'path' ) );
+        if ( selections.length < 1 ) {
+            warn( '`selectPrevious` triggered with no siblings' );
+            return;
+        }
 
-        if ( this.get( 'menu.pages' ) ) {
-            this.showContent();
+        const selection = selections.objectAt( selections.length - 1 );
+        const currentItems = Ember.get( selection, 'items' );
 
-            if ( !this.get( 'useDrillDownKey' ) ) {
-                this.set( 'keyHandler', true );
-                this.get( 'parentView' ).set( 'keyHandler', false );
-            }
-        } else {
-            const action = this.get( 'menu.action' );
-            const route = this.get( 'menu.route' );
-
-            if ( action ) {
-                if ( 'function' === Ember.typeOf( action ) ) {
-                    action.call( this );
-                } else if ( 'object' === Ember.typeOf( action ) ) {
-                    rootNode.sendAction(
-                        'actionInitiated',
-                        action.actionName,
-                        action.data
-                    );
-                } else {
-                    rootNode.sendAction( 'actionInitiated', action );
-                }
-            } else if ( route ) {
-                rootNode.sendAction( 'changeRoute', route );
-            }
+        if ( !currentItems ) {
+            throw new Ember.Error( 'Current items are undefined' );
+        }
 
-            rootNode.send( 'closeAll' );
+        if ( currentItems.length < 2 ) {
+            warn( '`selectPrevious` triggered with no siblings in context' );
+            return;
         }
+
+        const currentIndex = Ember.get( selection, 'index' );
+
+        if ( Ember.typeOf( currentIndex ) !== 'number' ) {
```

Comment:
Change order of compared values per https://github.com/softlayer/ember-style-guide/issues/18